config/src/main/java/com/alibaba/nacos/config/server/service/repository/embedded/BaseDatabaseOperate.java

```diff
@@ -213,18 +214,30 @@ default Boolean update(TransactionTemplate transactionTemplate, JdbcTemplate jdb
         return transactionTemplate.execute(status -> {
             String[] errSql = new String[] {null};
             Object[][] args = new Object[][] {null};
+            List<Boolean> listResult = new ArrayList<>(contexts.size());
             try {
                 contexts.forEach(pair -> {
                     errSql[0] = pair.getSql();
                     args[0] = pair.getArgs();
+                    boolean necessary = pair.isNecessary();
                     LoggerUtils.printIfDebugEnabled(LogUtil.DEFAULT_LOG, "current sql : {}", errSql[0]);
                     LoggerUtils.printIfDebugEnabled(LogUtil.DEFAULT_LOG, "current args : {}", args[0]);
-                    jdbcTemplate.update(pair.getSql(), pair.getArgs());
+                    int update = jdbcTemplate.update(pair.getSql(), pair.getArgs());
+                    listResult.add(update > 0 || !necessary);
                 });
                 if (consumer != null) {
-                    consumer.accept(Boolean.TRUE, null);
+                    if (!listResult.stream().anyMatch(e -> !e)) {
+                        consumer.accept(Boolean.TRUE, null);
+                        return Boolean.TRUE;
+                    }
+                    consumer.accept(Boolean.FALSE, null);
+                    return Boolean.FALSE;
```

Comment:
if one operation fails ，the prev operations should rollback in a transaction context, only returning false may not rollback prev operations.